src/main/java/ua/lviv/iot/parkingServer/logic/ParkingSpotService.kt

```diff
@@ -0,0 +1,46 @@
+package ua.lviv.iot.parkingServer.logic
+
+import org.springframework.stereotype.Service
+import ua.lviv.iot.parkingServer.datastorage.ParkingSpotFileStore
+import ua.lviv.iot.parkingServer.model.ParkingSpot
+import java.util.concurrent.atomic.AtomicLong
+import javax.annotation.PostConstruct
+import javax.annotation.PreDestroy
+import ua.lviv.iot.parkingServer.logic.exception.EntityNotFoundException
+
+@Service
+class ParkingSpotService(private val parkingSpotFileStore: ParkingSpotFileStore) {
+    private val parkingSpots: MutableMap<Long, ParkingSpot> = HashMap()
```

Comment:
Your `id generator`(AtomicLing) is thread safe now, awesome!

But 
```
private val parkingSpots: MutableMap<Long, ParkingSpot> = HashMap()
```
isn't thread safe.

The simplest fix is changing to `ConcurrentHashMap`. But there are some additional pitfalls that should be counted on.
I'll tell about them during our next 1-to-1

Reply:
computeIfAbsent
computeIfPresent
compute